### PR TITLE
Ensure that trailers are not lost when making blocking calls.

### LINKED ISSUE #2036: Metadata trailers are not passed to Status*Exception when using ClientCalls.blockingUnaryCall.
Trying to write a sample for #1295, but when calling unary rpc method on blocking stub, the trailers are not passed during `ClientCalls.getUnchecked` which is called from [ClientCalls.blockingUnaryCall](https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/ClientCalls.java#L141).

Is this by design or bug ?

Sample code is [here](https://github.com/chikei/grpc-java/commit/ea1ec3f5503a40d39c0ad10360c8a6ecbad46eca), the [blockingCall()](https://github.com/chikei/grpc-java/blob/ea1ec3f5503a40d39c0ad10360c8a6ecbad46eca/examples/src/main/java/io/grpc/examples/errorhandling/ErrorHandlingClient.java#L94) fails but others are good.

### PR DESCRIPTION
**Pull request by nmittler**
Fixes #2036

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 PTAL

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 1]
```diff
@@ -558,7 +558,7 @@ public void onClose(Status status, Metadata trailers) {
     /**
      * Waits until there is a Runnable, then executes it and all queued Runnables after it.
      */
-    public void waitAndDrain() throws InterruptedException {
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Can we avoid these unrelated changes (here and above)? I actually put the public there on purpose, although I agree removing it works, too.

File: core/src/main/java/io/grpc/StatusRuntimeException.java [hunk 2]
```diff
@@ -66,4 +68,27 @@ public final Status getStatus() {
...
   }
+
+  /**
+   * Wraps the given {@link Throwable} in a {@link StatusRuntimeException}. If it contains an
+   * embedded {@link StatusException} or {@link StatusRuntimeException}, the returned exception will
+   * contain the embedded trailers and status, with the given exception as the cause. Otherwise, an
+   * exception will be generated from an {@link Status#UNKNOWN} status.
+   */
+  @ExperimentalApi
+  public static StatusRuntimeException fromThrowable(Throwable t) {
```
Comment (ejona86) [inline core/src/main/java/io/grpc/StatusRuntimeException.java]: I'd rather this not be public. I can't imagine a case our users would need this. It seems stub-specific.

File: stub/src/main/java/io/grpc/stub/ClientCalls.java (hunk 1 above)
Comment (nmittler) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Sure ... was just fixing annoying warnings that were popping up in IntelliJ

File: stub/src/test/java/io/grpc/stub/ClientCallsTest.java [hunk 3]
```diff
@@ -74,17 +66,29 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Semaphore;
 import java.util.concurrent.TimeUnit;
+import org.junit.After;
```
Comment (ejona86) [inline stub/src/test/java/io/grpc/stub/ClientCallsTest.java]: This is going to break checkstyle. Getting onto the new import scheme is https://github.com/grpc/grpc-java/issues/2038

Comment (nmittler) [inline stub/src/test/java/io/grpc/stub/ClientCallsTest.java]: ack.

File: core/src/main/java/io/grpc/StatusRuntimeException.java (hunk 2 above)
Comment (nmittler) [inline core/src/main/java/io/grpc/StatusRuntimeException.java]: Where would you like it? ClientCalls?

Comment (ejona86) [inline core/src/main/java/io/grpc/StatusRuntimeException.java]: If it were package-private in client-calls that would be fine. I'd also be okay if it was in `internal` in `GrpcUtil`. I'm not bothered too much where, just as long as it does not appear at all like a public interface (even if unstable). So I guess even here would be fine if it were `@Internal`, but it is probably better to be more out-of-sight of our users (which both `internal` and package-private do).

**Comment by nmittler** [comment]
@ejona86 PTAL

**Comment by ejona86** [comment]
@nmittler LGTM. Please also backport to v1.0.x

**Comment by nmittler** [comment]
Cherry-picked into master as 780b269
